zap/src/zap.py

```diff
@@ -189,16 +190,26 @@ def get_gcp_token() -> str:
     """
     Generate a Google access token with custom scopes for the default identity.
     """
-    credentials, _ = google.auth.default(
-        scopes=[
-            "profile",
-            "email",
-            "openid",
-            "https://www.googleapis.com/auth/cloud-billing",
-        ]
-    )
-    credentials.refresh(GoogleAuthRequest())
-    return credentials.token
+    try:
```

Comment:
Catching this exception is good, but it means that you need a check in the code for if this worked or not. If this doesn't return a token there's no consequence in most of the code paths. Meaning a scan could continue without a token and run unauthenticated.
You could either raise another exception, or put a check around line 324 to see if the token is none and stoping the scan with a slack alert if the token doesn't exist.

Reply:
Yes I like the latter option.